File: mobile/ios/Sources/Data/TelemetryRepositoryImpl.swift
import Foundation
import AppDomain
import PantherSecurity

public final class TelemetryRepositoryImpl: TelemetryRepository {
    private let sdk: PantherSecuritySDK

    public init(sdk: PantherSecuritySDK = .shared) {
        self.sdk = sdk
    }

    public func send(event: TelemetryEvent) async throws {
        let request = PantherSecurityTelemetryRequest(
            eventId: event.eventId,
            appId: event.appId,
            appVersion: event.appVersion,
            env: event.env,
            device: PantherSecurityDeviceInfo(platform: "ios", osVersion: "iOS", model: "iPhone"),
            signals: PantherSecurityIntegritySignals(
                jailbreak: event.signals.jailbreak,
                root: event.signals.root,
                debugger: event.signals.debugger,
                hooking: event.signals.hooking,
                proxyDetected: event.signals.proxyDetected
            ),
            action: PantherSecurityActionContext(name: event.action.name, context: event.action.context),
            timestamp: event.timestamp,
            signature: "stub-signature"
        )

        try await sdk.sendTelemetry(request)
    }
}

File: mobile/android/app/src/main/java/com/panthersecurity/sample/data/TelemetryRepositoryImpl.kt
package com.panthersecurity.sample.data

import com.panthersecurity.sample.domain.TelemetryEvent
import com.panthersecurity.sample.domain.TelemetryRepository
import com.panthersecurity.sample.sdk.PantherSecurityActionContext
import com.panthersecurity.sample.sdk.PantherSecurityIntegritySignals
import com.panthersecurity.sample.sdk.PantherSecuritySdk
import com.panthersecurity.sample.sdk.PantherSecurityTelemetry

class TelemetryRepositoryImpl(private val sdk: PantherSecuritySdk) : TelemetryRepository {
    override suspend fun send(event: TelemetryEvent) {
        val request = PantherSecurityTelemetry(
            eventId = event.eventId,
            appId = event.appId,
            appVersion = event.appVersion,
            env = event.env,
            action = PantherSecurityActionContext(event.action.name, event.action.context),
            signals = PantherSecurityIntegritySignals(
                jailbreak = event.signals.jailbreak,
                root = event.signals.root,
                debugger = event.signals.debugger,
                hooking = event.signals.hooking,
                proxyDetected = event.signals.proxyDetected
            ),
            timestamp = event.timestamp
        )
        sdk.sendTelemetry(request)
    }
}
